test/python/circuit/test_circuit_load_from_qpy.py

```diff
@@ -261,3 +262,22 @@ def test_multiple_circuits(self):
         qpy_file.seek(0)
         new_circs = load(qpy_file)
         self.assertEqual(circuits, new_circs)
+
+    def test_standalone_register(self):
+        """Test a circuit with standalone registers."""
+        qubits = [Qubit() for _ in range(5)]
+        qc = QuantumCircuit()
+        qc.add_bits(qubits)
+        qr = QuantumRegister(bits=qubits)
+        qc.add_register(qr)
+        qc.h(qr)
+        qc.cx(0, 1)
+        qc.cx(0, 2)
+        qc.cx(0, 3)
+        qc.cx(0, 4)
+        qc.measure_all()
+        qpy_file = io.BytesIO()
+        dump(qc, qpy_file)
+        qpy_file.seek(0)
+        new_qc = load(qpy_file)[0]
+        self.assertEqual(qc, new_qc)
```

Comment:
Some additional test cases covering the cases where shared and standalone registers are used in a single circuit, or where registers overlap, would be helpful here.

Reply:
I added some more test cases in https://github.com/Qiskit/qiskit-terra/pull/6523/commits/3ddf3e0d52e8687f8bb24c8dfec22401c2435626 and https://github.com/Qiskit/qiskit-terra/pull/6523/commits/5d19aff790ea196d302f506c473055ab93ee2a5f do you have any other ones you think would be useful?